### PR TITLE
Remove NET461 leftovers and multi-target net5

### PR DESCRIPTION
**Pull request by Hawxy**
There's a bunch of stuff in here:

- Drop remaining net461 support items and cleanup csprojs
- Fix tests that fail if you live east of UTC
- Multi-target net5 to enable new roslyn analyzers and let contributors poke a stick at C# 9 features
- Multi-target test projects to cover range of supported .NET Core versions
- Port the .NET 5 reflection fix from 3.x
- Add a note to docs regarding compatibility
- Update dependencies across the board (fixes some weirdness with the VS test runner)

### CODE REVIEW

**Comment by lahma** [comment]
Just wanted to say thank you for your work and especially,

> Fix tests that fail if you live east of UTC

Cheers from Finland 🚀

File: src/Marten.Testing/Acceptance/document_inserts.cs [hunk 1]
```diff
@@ -51,6 +52,26 @@ public void can_insert_a_mixed_bag_of_documents()
                 query.Query<User>().Count().ShouldBe(4);
             }
         }
+#if NET
```
Comment (mysticmind) [inline src/Marten.Testing/Acceptance/document_inserts.cs:55]: addictive aspiration 🥇

File: README.md [hunk 2]
```diff
@@ -18,15 +18,11 @@ Marten also provides .NET developers with an ACID-compliant event store with use
 
 Before getting started you will need the following in your environment:
 
-**1. .NET Core SDK 3.1 (or higher)**
+**1. .NET Core SDK 5.0 (or higher)**
```
Comment (mysticmind) [inline README.md:21]: Should this include .NET Core SDK 3.1 as well?

**Review by mysticmind — APPROVED** [review]
Looks good to me 👍 . Few minor comments, please have a look.

File: README.md (hunk 2 above)
Comment (Hawxy) [reply README.md:21]: Not required, .NET Core SDKs are backwards compatible with lower targets.

Comment (Hawxy) [reply README.md:21]: I think the only exception to that would be for pointing the tests at the specific runtime version.

Comment (mysticmind) [reply README.md:21]: we can leave it as `.NET Core SDK 3.1 (or higher)`, this also means that it includes net5. thoughts?

Comment (Hawxy) [reply README.md]: I adjusted the wording to be more accurate

**Review by oskardudycz — APPROVED** [review]
Great stuff @Hawxy! Once again thank you for keeping an eye on the .NET 5 compatibility - really appreciated! 🙇

**Review by jeremydmiller — APPROVED** [review]
Looks good to me. Good catch w/ the Datetime to DateTimeOffset conversions in the tests

**Pull request merged**